jobs/src/main/scala/dcos/metronome/jobrun/impl/JobRunExecutorActor.scala

```diff
@@ -107,9 +108,13 @@ class JobRunExecutorActor(
   }
 
   def addTaskToLaunchQueue(): Unit = {
-    log.info("addTaskToLaunchQueue")
-    import dcos.metronome.utils.glue.MarathonImplicits._
-    launchQueue.add(jobRun.toRunSpec, count = 1)
+    if (existsInLaunchQueue()) {
```

Comment:
IIUC the issue was the the initial store timed out (b/c zk was slow?), the actor was restarted, tried to add to the launchQueue again, which failed because the initial store actually persisted the node.

Could you add a comment explaining why this could be the case? Not this particular scenario, but maybe something like _if the actor was restarted due to an exception, we have to check whether the jobRun was added to the launchQueue before already._